apps/antalmanac/src/lib/enrollmentHistory.ts

```diff
@@ -0,0 +1,154 @@
+            return latestEnrollmentHistory;
+        }
+
+        return null;
+    };
+
+    /**
+     * This function parses enrollment history data from PeterPortal so that
+     * we can pass the data into a recharts graph. For each element in the given
+     * array, merge the dates, totalEnrolledHistory, maxCapacityHistory,
+     * and waitlistHistory arrays into one array that contains the enrollment data
+     * for each day.
+     *
+     * @param res an array of enrollment histories from PeterPortal
+     * @returns an array of enrollment histories that we can use for the graph
+     */
+    parseEnrollmentHistoryResponse = (res: EnrollmentHistoryGraphQL[]): EnrollmentHistory[] => {
+        const parsedEnrollmentHistory: EnrollmentHistory[] = [];
+
+        for (const enrollmentHistory of res) {
+            const enrollmentDays: EnrollmentHistoryDay[] = [];
+
+            for (const [i, date] of enrollmentHistory.dates.entries()) {
+                const d = new Date(date);
+                const formattedDate = `${d.getMonth() + 1}/${d.getDate() + 1}/${d.getFullYear()}`;
+
+                enrollmentDays.push({
+                    date: formattedDate,
+                    totalEnrolled: Number(enrollmentHistory.totalEnrolledHistory[i]),
+                    maxCapacity: Number(enrollmentHistory.maxCapacityHistory[i]),
+                    waitlist:
+                        enrollmentHistory.waitlistHistory[i] === 'n/a'
+                            ? null
+                            : Number(enrollmentHistory.waitlistHistory[i]),
+                });
+            }
+
+            parsedEnrollmentHistory.push({
+                year: enrollmentHistory.year,
+                quarter: enrollmentHistory.quarter,
+                department: enrollmentHistory.department,
+                courseNumber: enrollmentHistory.courseNumber,
+                days: enrollmentDays,
+                instructors: enrollmentHistory.instructors,
+            });
+        }
+
+        return parsedEnrollmentHistory;
+    };
+
+    /**
+     * This function sorts the given array of enrollment histories so that
+     * the most recent quarters are in the beginning of the array.
+     *
+     * @param enrollmentHistory an array where each element represents the enrollment
+     * history of a course section during one quarter
+     */
+    sortEnrollmentHistory = (enrollmentHistory: EnrollmentHistory[]) => {
+        enrollmentHistory.sort((a, b) => {
+            const aTerm = `${a.year} ${a.quarter}`;
+            const bTerm = `${b.year} ${b.quarter}`;
+            // If the term for a appears earlier than the term for b in the list of
+            // term short names, then a must be the enrollment history for a later quarter
+            return this.termShortNames.indexOf(aTerm) - this.termShortNames.indexOf(bTerm);
+        });
+    };
+}
+
+const enrollmentHistoryCache = new _EnrollmentHistory();
+export default enrollmentHistoryCache;
```

Comment:
Personally, I don't really see the value in this pattern. It's essentially a singleton class that could be effectively captured within a module scope and functions. But this pattern already exists in the codebase, so the precedent has already been established. 

I think it would be more appropriate to mark most of the properties static.
For example, most of the class could be static -- including the cache, and you could initialize special instances that will write to the static cache.

```tsx
class EnrollmentHistory {
  static cache: Record<string, any> = {}
  department: string
  partialQueryString: string
  constructor(department: string) {
    this.department = department
    this.partialQueryString = `...`
  }

  find(course: string) {
    // (not real, just example)
    const fullQueryString = this.partialQueryString + course
    return EnrollmentHistory.cache[fullQueryString] ?? this.findAndCache(course)
  }

  findAndCache(course: string) {
    // ...
  }
}

export const CsEnrollmentHistory = EnrollmentHistory("CS")
```
This makes sense to me, because individual instances of the classes serve the distinct purpose of searching and caching for a specific department. And they all write to a shared static cache. This is at your discretion as the implementer.